packages/arduino-cli/src/cpuParser.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * One bright day this module will be burned.
+ * See: https://github.com/arduino/arduino-cli/issues/45
+ */
+
+import path from 'path';
+import * as R from 'ramda';
+import * as fse from 'fs-extra';
+import promiseAllProperties from 'promise-all-properties';
+
+const PACKAGES_DIR = 'packages';
+const HARDWARE_DIR = 'hardware';
+const BOARDS_FNAME = 'boards.txt';
+
+// :: Path -> FQBN -> String -> Path
+export const getBoardsTxtPath = R.curry((dataPath, fqbn, version) => {
+  const [packageName, archName] = R.split(':', fqbn);
+  return path.resolve(
+    dataPath,
+    PACKAGES_DIR,
+    packageName,
+    HARDWARE_DIR,
+    archName,
+    version,
+    BOARDS_FNAME
+  );
+});
+
+/**
+ * Parses Arduino's `.txt` definition file. Like `boards.txt` or `platform.txt`.
+ *
+ * In the reduce function we have to store value of `A.menu.cpu.B` into special
+ * property to avoid breaking it with associating other properties.
+ * E.G.
+ * `nano.menu.cpu.atmega328 = ATmega328`
+ * stores String 'ATmega328' by the path `menu.cpu.atmega328`
+ * but then config has next lines:
+ * `nano.menu.cpu.atmega328.upload.speed = 12800`
+ * so our string will be converted into object in which will be associated
+ * object `upload` with property `speed`.
+ * As a result we have something like this:
+ * ```
+ *   {
+ *     menu: {
+ *       cpu: {
+ *         atmega328: {
+ *           0: 'A',
+ *           1: 'T',
+ *           2: 'm',
+ *           ...
+ *           upload: {
+ *             speed: '12800'
+ *           }
+ *         }
+ *       }
+ *     }
+ *   }
+ * ```
+ * To prevent it we'll store `nano.menu.cpu.atmega328 = ATmega328` into
+ * special property `cpuName` and then we have:
+ * ```
+ * {
+ *   atmega328: {
+ *     cpuName: 'ATmega328',
+ *     upload: {
+ *       speed: '12800'
+ *     }
+ *   }
+ * }
+ * ```
+ */
+// :: String -> Object
+const parseTxtConfig = R.compose(
+  R.reduce(
+    (acc, [tokens, value]) =>
+      R.compose(
+        R.assocPath(R.__, value, acc),
+        R.when(
+          R.both(R.propEq(1, 'menu'), R.compose(R.equals(4), R.length)),
+          R.append('cpuName')
+        )
+      )(tokens),
+    {}
+  ),
+  R.map(R.compose(R.zipWith(R.call, [R.split('.'), R.identity]), R.split('='))),
```

Comment:
What do you mean? :)

Reply:
I guess @evgenykochetkov brain has exploded trying to parse the line. Mine too.

Reply:
It's a dumb copy from `xod-arduino-deploy` and it works fine.
But in the next PR it will be replaced with brand-new `optionParser`, so you can just trust me that works fine 🤣